contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -4648,8 +4729,30 @@ get_bbf_pivot_tuplestore(RawStmt 	*sql,
 		/* internal error */
 		elog(ERROR, "get_bbf_pivot_tuplestore: SPI_connect returned %d", ret);
 
-	/* Now retrieve the bbf_pivot source rows */
-	ret = SPI_execute_raw_parsetree(sql, sourcetext, true, 0);
+	/* 
+	 * Now retrieve the bbf_pivot source rows    
+	 *
+	 * If view_uuid is defined, then we will use the provided uuid to form the source_sql and 
+	 * get result. source_sql name format is pvt_sv_[uuid]
+	 * 
+	 * if view_uuid is not defined, then we will use the rawparsetree of source_sql to get the
+	 * result
+	 */
+	if (!view_uuid || strlen(view_uuid) != 0)
+	{
+		StringInfoData 		buf;
+		
+		initStringInfo(&buf);
+		appendStringInfoString(&buf, "select * from ");
+		appendStringInfoString(&buf, BBF_PIVOT_VIEW_SRC_PREFIX);
+		appendStringInfoString(&buf, view_uuid);
```

Comment:
nit: where is `view_uuid` value coming from? is  it possible for it to have space in the beginning?

Reply:
catalog.c:6122

Reply:
no space is generated in the beginning